src/main.rs

```diff
@@ -208,54 +208,74 @@ mod test_song_segments {
 	}
 }
 
-pub enum FileFormat {
+pub enum FileType {
 	SegmentFormat,
 	SongArchiveFormat,
-	InvalidFormat,
 }
 
-pub fn detect_file_type(path: &Path) -> FileFormat {
-	let extension = path.file_name().unwrap().to_str().unwrap().split(".").last().unwrap();
+pub fn detect_file_type(file_name: &str) -> Result<FileType, AppError> {
+	let extension = file_name.split(".").last().unwrap();
 	match extension {
-		"wav" | "ogg" | "mp3" | "flac" => FileFormat::SegmentFormat,
-		"zip" => FileFormat::SongArchiveFormat,
-		_ => FileFormat::InvalidFormat,
+		"wav" | "ogg" | "mp3" | "flac" => Ok(FileType::SegmentFormat),
+		"zip" => Ok(FileType::SongArchiveFormat),
+		_ => Err(AppError::UnrecognizedSongFormat(file_name.to_string())),
```

Comment:
Instead of straight up failing when we encounter an unexpected file, we should just ignore the file.

Reply:
It already doesn't, check the call site of `detect_file_type` (in `initialize_songs`).

Reply:
Ah you're right